sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2953,11 +2953,20 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
     zePrint("piEnqueueKernelLaunch: unsupported work_dim\n");
     return PI_INVALID_VALUE;
   }
-
-  assert(GlobalWorkSize[0] == (ZeThreadGroupDimensions.groupCountX * WG[0]));
-  assert(GlobalWorkSize[1] == (ZeThreadGroupDimensions.groupCountY * WG[1]));
-  assert(GlobalWorkSize[2] == (ZeThreadGroupDimensions.groupCountZ * WG[2]));
-
+  
+  if(GlobalWorkSize[0] != (ZeThreadGroupDimensions.groupCountX * WG[0])) {
+    zePrint("piEnqueueKernelLaunch: invalid work_dim\n");
+    return PI_INVALID_WORK_GROUP_SIZE;
+  }
+  if(GlobalWorkSize[1] != (ZeThreadGroupDimensions.groupCountY * WG[1])) {
+    zePrint("piEnqueueKernelLaunch: invalid work_dim\n");
+    return PI_INVALID_WORK_GROUP_SIZE;
+  }
+  if(GlobalWorkSize[2] != (ZeThreadGroupDimensions.groupCountZ * WG[2])) {
+    zePrint("piEnqueueKernelLaunch: invalid work_dim\n");
```

Comment:
Please add some helpful message for the end-user, like:
```suggestion
    zePrint("piEnqueueKernelLaunch: invalid work_dim: the range is not a multiple of the group size in the 3rd dimension\n");
```
Add the same for the  other dimensions.

Reply:
@keryell ,Thank you for the feedback. I have incorporated the proposed changes.